### DISCUSSION TITLE
Add better handling of WhatsApp Status updates

### DISCUSSION THREAD

**Discussion by samtrek** [d99kris/nchat#137] [Ideas]
I observed today that status shared by users now come as a personal message `nchat v.36, git:g2e00a7c`, while this is good, as nchat users can now reply to such statuses. However nchat users cannot distinguish between status and normal messages this can lead to replying to message that is meant generally for the public, or can be be disruptive as some people tends to share lots of statues which can push current conversation far below new statues.
Suggested solutions:
1. Users should be able to disable and enable if they want to see statuses.
2. A special marker to show that the new message is a status and and should not be mistaken as direct message.
3. Dont push chats with status message to the top of the chat, leave them as they are on the contact list window during chat.
 An option can be created to disable that new status message should be pushed to head of the contact list or should maintain its current position on the current list.
4. Or simply revert to the previous implementation as it shows what the status message are for, although nchat whatsapp users will not be able to reply to such status messages.

I dont know what it will take to implement all this but waiting for your take on it. keep up the good work

**Comment by d99kris** [comment]
Hi @samtrek - thanks for the feedback! I unfortunately (or fortunately 🙂) have no contacts sharing status updates so I had
missed this fact when I implemented the new handling of broadcast messages which put them in the chats of their respective
sender.

I am preparing a fix which will group status updates under a dedicated chat conversation `Status Updates`, this will hopefully minimize user confusion about them. WhatsApp's official app also shows them separately (in a dedicated **tab**, but still).

Later we can also consider allowing the user to mute the `Status Updates` chat.

Broadcast messages will continue go into the conversation of the sender (no change).

**Reply by d99kris** [reply]
Grouping of all WhatsApp Status updates under a single dedicated chat was implemented in a8bee5c.

For hiding/muting of `Status Updates` conversation, it will be tracked under #146.

Will proceed to close this discussion.
